### PR TITLE
Feature/fetch post recipes

### LINKED ISSUE #51: POST saved Recipes

### LINKED ISSUE #53: Optional - Delete saved recipes (fetch)

### PR DESCRIPTION
**Pull request by jalbe0076**
### What is the feature?
- This PR will close #51  & close #53
- Added fetch post function to post user recipes to the current user on the local server
- Added a new function to reformat the user recipes to cook list into recipe objects instead of ID to keep current code functionality
- Added fetch delete function to delete the recipes from the local server
- Data will persist on page load

### How you implemented the solution?
- used fetch POST and fetch DELETE

### Does it impact any other area of the project?
- Several functions had to be refactored to modify user recipes to cook list
- added edge cases to change the heart colour when saving recipes

### Any changes in the UI (Screenshots)?
- none

### How to test the feature (A test scenario or any new setup)?
- `git pull` and `git fetch` from main to pull the branch to your local machine, checkout into the `feature/fetch-post-recipes` branch and test the web page while running the local server. To setup the local server clone the following repo and follow installation procedures https://github.com/turingschool-examples/whats-cookin-api

### CODE REVIEW

**Comment by jalbe0076** [comment]
> Idk if it's just me or if not supposed to work yet - but the All Categories dropdown doesn't work when clicking on a category.

Good catch! I missed this when I was testing. Fixed the search bar which also did not work...

File: src/recipes.js [hunk 1]
```diff
@@ -108,6 +108,16 @@ const alphabetizeData = (data) => {
   return data;
 }
 
+const userRecipes = (user, recipes) => {
```
[the remark this answers is not in the document]
Comment (jalbe0076) [reply src/recipes.js]: yeah... done!

**Review by alivaditis — APPROVED** [review]
post be crazy, great job

**Review by CaliHam — APPROVED** [review]
Awesome stuff!!

**Review by paulina-isabel — APPROVED** [review]
